applications/StructuralMechanicsApplication/custom_response_functions/adjoint_elements/adjoint_finite_difference_spring_damper_element_3D2N.cpp

```diff
@@ -28,8 +28,19 @@ void AdjointFiniteDifferenceSpringDamperElement<TPrimalElement>::InitializeSolut
     // would be part of the element properties this transferring would be not necessary. The stiffness parameters
     // are needed by the primal element in order to compute later on the element stiffness matrix for the
     // adjoint problem and the sensitivity matrix as the element contribution to the pseudo-load.
-    this->pGetPrimalElement()->SetValue(NODAL_DISPLACEMENT_STIFFNESS, this->GetValue(NODAL_DISPLACEMENT_STIFFNESS));
-    this->pGetPrimalElement()->SetValue(NODAL_ROTATIONAL_STIFFNESS, this->GetValue(NODAL_ROTATIONAL_STIFFNESS));
+    auto stiffness_available = false;
+    if (this->Has(NODAL_DISPLACEMENT_STIFFNESS)) {
+        const auto temp = this->GetValue(NODAL_DISPLACEMENT_STIFFNESS);
+        this->pGetPrimalElement()->SetValue(NODAL_DISPLACEMENT_STIFFNESS, temp);
+        stiffness_available = true;
+    }
+    if (this->Has(NODAL_ROTATIONAL_STIFFNESS)) {
+        const auto temp = this->GetValue(NODAL_ROTATIONAL_STIFFNESS);
+        this->pGetPrimalElement()->SetValue(NODAL_ROTATIONAL_STIFFNESS, temp);
+        stiffness_available = true;
+    }
+    KRATOS_ERROR_IF_NOT(stiffness_available) << "Neither NODAL_DISPLACEMENT_STIFFNESS nor NODAL_ROTATIONAL_STIFFNESS available!" << std::endl;
```

Comment:
I would do this check in `Check`

Reply:
I don't think that is possible. The stiffness values typically set to the elements by `AssignVectorVariableToEntitiesProcess` which executes within `InitializeSolutionStep` to my knowledge.

Reply:
ok then I would do the following: 

~~~c++
const auto& r_const_this = *this;
this->pGetPrimalElement()->SetValue(NODAL_DISPLACEMENT_STIFFNESS, r_const_this.GetValue(NODAL_DISPLACEMENT_STIFFNESS));
this->pGetPrimalElement()->SetValue(NODAL_ROTATIONAL_STIFFNESS, r_const_this.GetValue(NODAL_ROTATIONAL_STIFFNESS));

KRATOS_ERROR_IF_NOT(this->Has(NODAL_DISPLACEMENT_STIFFNESS) || this->Has(NODAL_ROTATIONAL_STIFFNESS))
~~~

Reply:
you are right. I modified it as proposed.